sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,129 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+FORTRAN_NEEDED=fortran
+FORTRAN_STANDARD=90
+
+inherit desktop fortran-2 python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti fortran pch +python test"
+
+REQUIRED_USE="
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	python? ( cti )
+	${PYTHON_REQUIRED_USE}
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+	sci-libs/sundials:0=
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
+	dev-libs/libfmt
+	python? (
+		dev-python/cython[${PYTHON_USEDEP}]
+	)
+	test? (
+		>=dev-cpp/gtest-1.8.0
+	)
+"
+
+PATCHES=( "${FILESDIR}/${PN}_${PV}_libdirname_variable.patch" )
+
+pkg_setup() {
+	fortran-2_pkg_setup
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	# patch to work 'scons test' properly in case of set up 'renamed_shared_libraries="no"' option
+	sed -i "s/, libs=\['cantera_shared'\]//" "${S}"/test_problems/SConscript || die "failed to modify 'test_problems/SConscript'"
+	# patch env to pass CCACHE_DIR variable
+	sed -i "s/ENV={'PATH': os.environ\['PATH'\]}/ENV={'PATH': os.environ\['PATH'\], 'CCACHE_DIR': os.environ.get('CCACHE_DIR','')}/" "${S}"/SConstruct || die "failed to modify 'SConstruct'"
+}
+
+## Full list of configuration options of Cantera is presented here:
+## http://cantera.org/docs/sphinx/html/compiling/config-options.html
+
+src_configure() {
+	scons_vars=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+		cc_flags="${CXXFLAGS}"
+		cxx_flags="-std=c++11"
+		debug="no"
+		FORTRAN="$(tc-getFC)"
+		FORTRANFLAGS="${CXXFLAGS}"
+		renamed_shared_libraries="no"
+		use_pch=$(usex pch)
+## In some cases other order can break the detection of right location of Boost: ##
+		system_fmt="y"
+		system_sundials="y"
+		system_eigen="y"
+		env_vars="all"
+		extra_inc_dirs="/usr/include/eigen3"
+	)
+	use test || scons_vars+=( googletest="none" )
+
+	scons_targets=(
+		f90_interface=$(usex fortran y n)
+		python2_package="none"
+	)
+
+	if use cti ; then
+		local scons_python=$(usex python full minimal)
+		scons_targets+=( python3_package="${scons_python}" python3_cmd="${EPYTHON}" )
+	else
+		scons_targets+=( python3_package="none" )
+	fi
+}
+
+src_compile() {
+	escons build "${scons_vars[@]}" "${scons_targets[@]}" prefix="/usr"
+}
+
+src_test() {
+	escons test
+}
+
+src_install() {
+	escons install stage_dir="${D%/}" libdirname="$(get_libdir)"
+	if ! use cti ; then
+		rm -r "${D%/}/usr/share/man" || die "Can't remove man files."
+	else
+	# Run the byte-compile of modules
```

Comment:
Misindent. Also, I suppose you can unset the `PYTHONDONTWRITEBYTECODE` thingie instead.

Reply:
I suppose that the right place of byte compiling is here because pyc files havr hardcoded mentioning paths inside of *.pyc files and otherwise in case of some errors it shows message that *.pyc was in file at `/var/tmp/portage/sci-libs/...`. Also as I set `stage_dir` via `${D}` in src_compile section it result in the recompiling of byte code again. So it's better to compile it once in at the end of src_install with right paths.

Sorry, i don't understand the phrase about Misindent. Should I start this comment at column 1 without intendation?

Reply:
The code inside `else` branch should be indented one level compared to `else`.

Reply:
>  Regenerating caches isn't necessary here, is it?

It seems that is. At least in my KDE the added items appeared and removed without updating the cache.

Then I shall replace the onherit xdg-utils with RDEPEND of package.
I use xdg-open as I didn't find another way to create the .desktop menu item to open html file.

Reply:
Done.
sci-libs/cantera: indentation is fixed.

app-doc/cantera-docs: `inherit xdg-utils` replaced with `RDEPEND="x11-misc/xdg-utils"`; `pkg_postinst()` and `pkg_postrm()` sections are removed.